Web/server.py

```diff
@@ -1,25 +1,20 @@
+import httpConnection
+import bankApplication
+
 import socket
+import logging
 
+logging.basicConfig(filename='server.log', level=logging.DEBUG)
 
 def main():
     serverSocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
     serverSocket.bind(("127.0.0.1", 8080))
     serverSocket.listen()
     clientSocket, addr = serverSocket.accept()
+    logging.debug(f"Client connected on IP {clientSocket.getpeername()}")
 
-    dataAux = b''
-
-    while True:
-        data = clientSocket.recv(1024)
-        if not data:
-            break
-
-        else:
-            dataAux = dataAux + data
-
-    print()
-    clientSocket.sendall(dataAux)
-    print(data.decode("utf-8"))
+    connection = httpConnection.httpConnection(clientSocket)
+    connection.process(bankApplication.root)
 
     clientSocket.close()
     serverSocket.close()
```

Comment:
if happens some exception on connection.process, this lines don't execute and a tcp port will not be freed, so use this sockets over a "with" statement

Reply:
Can I use a "with" statement inside another "with" statement?

Reply:
Yes